src/Symfony/Bundle/FrameworkBundle/DependencyInjection/FrameworkExtension.php

```diff
@@ -551,8 +551,12 @@ public function load(array $configs, ContainerBuilder $container)
         $container->registerForAutoconfiguration(LoggerAwareInterface::class)
             ->addMethodCall('setLogger', [new Reference('logger')]);
 
-        $container->registerAttributeForAutoconfiguration(AsEventListener::class, static function (ChildDefinition $definition, AsEventListener $attribute): void {
-            $definition->addTag('kernel.event_listener', get_object_vars($attribute));
+        $container->registerAttributeForAutoconfiguration(AsEventListener::class, static function (ChildDefinition $definition, AsEventListener $attribute, \ReflectionClass $reflectionClass, ?\ReflectionMethod $reflectionMethod): void {
+            $tagAttributes = get_object_vars($attribute);
+            if ($reflectionMethod) {
+                $tagAttributes['method'] = $reflectionMethod->getName();
```

Comment:
we should throw if the attribute is on a method and also defines a "method" entry

Reply:
Done